src/fsharp/pars.fsy

```diff
@@ -3771,15 +3770,47 @@ optRange:
   | declExpr DOT_DOT declExpr 
       { SynIndexerArg.Two(mkSynOptionalExpr (rhs parseState 1) (Some $1), mkSynOptionalExpr (rhs parseState 3) (Some $3)) }
 
-  | declExpr DOT_DOT 
+  | INFIX_AT_HAT_OP declExpr DOT_DOT INFIX_AT_HAT_OP declExpr 
```

Comment:
Is it possible to modify existing rules instead of adding new ones to account for `^`?

I think you should be able to create an optional rule for `^`:
```
opt_fromEnd:
    | INFIX_AT_HAT_OP { true  }
    | /* EMPTY */     { false }
```

Using `opt_fromEnd`, you can add it to existing rules. For example, the rule case on `optRange`:
```
| declExpr DOT_DOT declExpr
```
Could be changed to:
```
| opt_fromEnd declExpr DOT_DOT opt_fromEnd declExpr
```
If we are able to do this, it will play nicely into the `SynIndexerArg` tweaks.

Reply:
Ah yes, we could just pass opt_fromEnd into the constructor of SynIndexerArg. That would allow us to simplify our parsing logic by quite a bit.